lib/kube/proxy/server.go

```diff
@@ -560,12 +602,18 @@ func (t *TLSServer) getKubernetesServersForKubeClusterFunc() (getKubeServersByNa
 			return []types.KubeServer{srv}, nil
 		}, nil
 	case ProxyService:
-		return t.getAuthKubeServers, nil
+		return func(ctx context.Context, name string) ([]types.KubeServer, error) {
+			servers, err := t.KubernetesServersWatcher.GetKubeServersByClusterName(ctx, name)
+			return servers, trace.Wrap(err)
+		}, nil
 	case LegacyProxyService:
 		return func(ctx context.Context, name string) ([]types.KubeServer, error) {
+			// If this is a legacy kube proxy, then we need to return the local kube servers if
+			// the local server is proxying the target cluster, otherwise act like a proxy_service.
+			// and forward the request to the next proxy.
 			kube, err := t.getKubeClusterWithServiceLabels(name)
 			if err != nil {
-				servers, err := t.getAuthKubeServers(ctx, name)
+				servers, err := t.KubernetesServersWatcher.GetKubeServersByClusterName(ctx, name)
```

Comment:
`KubernetesServerWatcher` is nil for `kubernetes_service` since it uses its internal state and does not need it. The `CheckAndSetDefautls` ensures that it's not nil for Proxy and legacy proxy.

kubeCusterWatcher is enabled only if you specify the watcher labels in the kubernetes service. 99.9% of the cases, users don't.
We can use a no-op watcher but feels too much for me.

Let me know what you think